runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptionsTest.java

```diff
@@ -90,4 +93,39 @@ public void testUtf8UserNameAndApplicationNameIsNormalized() {
     options.setAppName("fəˈnɛtık əsoʊsiˈeıʃn");
     assertEquals("f00n0t0k00so0si0e00n-0i00nt00n000n0l0-1208190706", options.getJobName());
   }
+
+  @Test
+  public void testStagingLocation() {
+    DataflowPipelineOptions options = PipelineOptionsFactory.as(DataflowPipelineOptions.class);
+    IOChannelUtils.registerStandardIOFactories(options);
+    options.setTempLocation("file://temp_location");
+    options.setStagingLocation("gs://staging_location");
+    assertEquals("gs://staging_location", options.getStagingLocation());
+  }
+
+  @Test
+  public void testDefaultStagingLocation() {
+    DataflowPipelineOptions options = PipelineOptionsFactory.as(DataflowPipelineOptions.class);
+    IOChannelUtils.registerStandardIOFactories(options);
+    options.setTempLocation("gs://temp_location");
+    assertEquals("gs://temp_location/staging", options.getStagingLocation());
+  }
+
+  @Test
+  public void testDefaultStagingLocationInvalid() {
+    DataflowPipelineOptions options = PipelineOptionsFactory.as(DataflowPipelineOptions.class);
+    options.setTempLocation("file://temp_location");
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("GCP temp location requires a valid 'gs://' path");
+    options.getStagingLocation();
```

Comment:
You are right. The code and test are wrong. 
GcpTempLocation should be optional and not throw.
Updated